ingest/anndata_.py

```diff
@@ -35,3 +44,65 @@ def validate(self):
         except ValueError:
             return False
 
+    @staticmethod
+    def generate_cluster_header(adata, clustering_name):
+        """
+        Based on clustering dimensions, write clustering NAME line to file
+        """
+        dim2 = ['NAME', 'X', 'Y']
+        dim3 = ['NAME', 'X', 'Y', 'Z']
+        if adata.obsm[clustering_name].shape[1] == 2:
+            headers = dim2
+        elif adata.obsm[clustering_name].shape[1] == 3:
+            headers = dim3
+        elif adata.obsm[clustering_name].shape[1] > 3:
+            msg = f"Data in obsm \"{clustering_name}\" has too many dimensions for visualization."
+            # !!! is ValueError the right type?
+            raise ValueError(msg)
+        else:
+            msg = f"Data in obsm \"{clustering_name}\" has too few dimensions for visualization."
```

Comment:
Per other comment:

```suggestion
            msg = f"Too few dimensions for visualization in obsm \"{clustering_name}\"."
```

Maybe also useful to include:
* How many dimensions were found?
* How few is too few?

Reply:
`9bc7f61`